docs/testing.md

```diff
@@ -0,0 +1,116 @@
+# Testing Tools
+
+This document describes several techniques to test the code, especially before committing or submitting PRs.
+
+## Unit testing
+
+## Integration testing
+
+## Mutation testing
+
+### Overview
+
+Mutation testing is complementary to code coverage.
+It evaluates the tests' quality of the functions added or modified in the PR.
+Mutation testing involves making small changes (mutations) to the code to check if the tests can detect these changes.
+This mutations represent a clone of the original code with one given function having a default value instead of its body.
+Then the `cargo build` and `cargo nextest` for the package intended are called. If the mutated code breaks, this means that function is properly tested.
+
+This is useful for any size of changed code, being it one line in one function, or tens and hundreds of functions, but the time required increases directly with the number of mutants.
```

Comment:
I would prefer to keep this simple - perhaps a link to wikipedia etc about what mutation testing is?